pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -237,7 +237,8 @@ func (c *Controller) etcdRemoveSnapshotWithBackoff(b *v3.EtcdBackup) error {
 		return err
 	}
 	return wait.ExponentialBackoff(backoff, func() (bool, error) {
-		if inErr := c.backupDriver.ETCDRemoveSnapshot(c.ctx, cluster.Name, kontainerDriver, cluster.Spec, b.Name); inErr != nil {
+		snapshotName := clusterprovisioner.GetBackupFilename(b)
```

Comment:
Can we move line#240 out of the ExponentialBackoff loop so it is invoked only once?